### PR TITLE
Feature/vih 4532 back functionality

### PR DESCRIPTION
**Pull request by rehmanab**
https://tools.hmcts.net/jira/browse/VIH-4532

Store to session storage the model so when F5, loads saved data

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/representative-journey/services/representative.journey.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    representativeJourneyService = new RepresentativeJourneyService();
+  });
+
+  it('returns null when item not exist', () => {
+    expect(representativeJourneyService.get()).toBeNull();
+  });
+
+  it('returns the model from storage', () => {
+    const model = new MutableRepresentativeSuitabilityModel();
+    model.hearing = new Hearing('hearing1', new Date(2012, 12, 12));
```
Comment (ddikman) [inline ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/representative-journey/services/representative.journey.service.spec.ts]: test with date and time?

File: ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/representative-journey/services/representative.journey.service.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  constructor() {
+    this.cache = new SessionStorage<RepresentativeSuitabilityModel>('REPRESENTATIVEJOURNEY_MODEL');
+  }
+
+  get(): RepresentativeSuitabilityModel {
+    const response = this.cache.get();
+
+    if (response === null) { return null; }
+
+    const model = new MutableRepresentativeSuitabilityModel();
```
Comment (ddikman) [inline ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/representative-journey/services/representative.journey.service.ts]: might be worthwhile to do a quick comment on why you need to do this despite the `response` seemingly being an instance of `RepresentativeSuitabilityModel` already. you and I know what the issue was about but a few weeks down the road we'll be like "what's this all about?"

File: ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/shared/services/session-storage.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ export class SessionStorage<TType> {
      */
     get(): TType {
         const cached = sessionStorage.getItem(this.storageKey);
-        if (!cached) {
+        if (!cached || cached === undefined || cached === 'undefined') {
```
Comment (ddikman) [inline ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/shared/services/session-storage.ts]: this doesn't quite look right. we shouldn't allow to store it as 'undefined' it must be a bug somewhere. maybe better check in the `set()` method that we don't allow `null` or `undefined` to be set?

then `!cached` should be enough to raise if it's not set.

File: ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/representative-journey/services/representative.journey.service.spec.ts (hunk 1 above)
Comment (rehmanab) [reply ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/representative-journey/services/representative.journey.service.spec.ts]: done

File: ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/representative-journey/services/representative.journey.service.ts (hunk 2 above)
Comment (rehmanab) [reply ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/representative-journey/services/representative.journey.service.ts]: done

File: ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/shared/services/session-storage.ts (hunk 3 above)
Comment (rehmanab) [reply ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/shared/services/session-storage.ts]: I see your point, but we can't control everyone from not having a bug and storing undefined. As we can reliably say that no one will ever WANt to store Undefined, we can defend here for occasions where it is stored... What you think ?

Comment (rehmanab) [reply ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/shared/services/session-storage.ts]: Changed to check for undefined when trying to set the cached value

**Pull request merged**